server/client.go

```diff
@@ -1350,13 +1350,13 @@ func (c *client) closeConnection() {
 		}
 
 		if rid != "" && srv.remotes[rid] != nil {
-			Debugf("Not attempting reconnect for solicited route, already connected to \"%s\"", rid)
+			c.srv.Debugf("Not attempting reconnect for solicited route, already connected to \"%s\"", rid)
```

Comment:
I don't think so in practice, only in unit tests when we are manually constructing `client` literals for stubbing purposes. AFAIK [here](https://github.com/nats-io/gnatsd/blob/d6069447d22cc48d7b70c144f4634784748852bb/server/route.go#L317) and [here](https://github.com/nats-io/gnatsd/blob/d6069447d22cc48d7b70c144f4634784748852bb/server/server.go#L551) are the only places we construct a `client`, and in both places the server is non-nil.